services/criterion/criterion.service.js

```diff
@@ -0,0 +1,70 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const { BaseJsonService } = require('..')
+
+const IMPROVED = 'improved'
+
+function errorMessagesFor(notFoundMessage = 'status not found') {
+  return {
+    404: notFoundMessage,
+    422: notFoundMessage,
+  }
+}
+
+const schema = Joi.string().required()
+
+module.exports = class Criterion extends BaseJsonService {
+  static get category() {
+    return 'analysis'
+  }
+
+  static get route() {
+    return {
+      base: 'criterion',
+      pattern: ':user/:repo',
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Criterion',
+        namedParams: {
+          user: 'chmoder',
+          repo: 'data_vault',
+        },
+        staticPreview: this.render({ isImproved: true, status: IMPROVED }),
+      },
+    ]
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'criterion' }
+  }
```

Comment:
We've recently moved to leveraging static fields now that our prod runtime is on a new enough version of Node. We're going to have a lot of existing classes that need to be converted, but let's go ahead and leverage the new paradigm with this new service class

```suggestion
  static category = 'analysis'
  static route = { base: 'criterion', pattern: ':user/:repo' }
  static examples = [
    {
          title: 'Criterion',
          namedParams: {
            user: 'chmoder',
            repo: 'data_vault',
          },
          staticPreview: this.render({ isImproved: true, status: IMPROVED }),
      },
  ]
  
  static defaultBadgeData = { label: 'criterion' }
```

Reply:
👍